lambda/translprim.ml

```diff
@@ -1432,7 +1432,9 @@ let specialize_primitive env loc ty ~has_constant_constructor prim =
       Some (Comparison(comp, Compare_ints))
     end else if (is_base_type env p1 Predef.path_int
         || is_base_type env p1 Predef.path_char
-        || (maybe_pointer_type env p1 = Immediate)) then begin
+        || ((* Non-null external types are always represented by tagged integers. *)
+            maybe_pointer_type env p1 = Immediate
```

Comment:
I think I would prefer if the return type of `maybe_pointer` was tweaked to carry nullability information. In my mind, `Immediate` implies `Non_null`, so if this is not the case I'd like to be more explicit about it

Reply:
Good idea. This caught a case I missed with `[@untagged]` annotations on externals, fixed and added a test.